tests/test_port_config.py

```diff
@@ -154,6 +154,78 @@ def test_port_breakout(self, dvs, port_config):
             assert hw_lane_value, "Can't get hw_lane list"
             assert hw_lane_value == "1:%s" % (new_lanes[i])
 
+    def test_recirc_port(self, dvs):
+
+        # Get port config from configDB
+        cfg_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+        cfg_port_tbl = swsscommon.Table(cfg_db, swsscommon.CFG_PORT_TABLE_NAME)
+
+        keys = cfg_port_tbl.getKeys()
+        for port in keys:
+            (status, fvs) = cfg_port_tbl.get(port)
+            assert(status == True)
+
+        indexes = []
+        lanes = []
+        for fv in fvs:
+            if fv[0] == "index":
+                indexes.append(int(fv[1]))
+            if fv[0] == "lanes":
+                lanes.extend([int(lane) for lane in fv[1].split(",")])
+
+        # Stop swss before modifing the configDB
+        dvs.stop_swss()
+        time.sleep(1)
+
+        recirc_port_lane_base = max(lanes) + 1
+        recirc_port_index_base = max(indexes) + 1
+
+        # Add recirc ports to port config in configDB
+        recirc_port_lane_name_map = {}
+        for i in range(2):
+            name = alias = "Recirc%s" % i
+            fvs = swsscommon.FieldValuePairs([("role", "Rec"),
+                                              ("alias", alias),
+                                              ("lanes", str(recirc_port_lane_base + 1)),
+                                              ("speed", "10000"),
+                                              ("index", str(recirc_port_index_base + 1))])
+            cfg_port_tbl.set(name, fvs)
+
+        # Start swss
+        dvs.start_swss()
+        time.sleep(5)
+
+        app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+        app_port_tbl = swsscommon.Table(app_db, swsscommon.APP_PORT_TABLE_NAME)
+
+        # Verify recirc ports in port table in applDB
+        for i in range(2):
+            name = alias = "Recirc%s" % i
+            status, fvs = app_port_tbl.get(name)
+            assert status == True
+            fvs = {fv[0] : fv[1] for fv in fvs}
+            assert (fvs["role"] == "Rec" and
+                    fvs["alias"] == name and
+                    fvs["lanes"] == str(recirc_port_lane_base + 1) and
+                    fvs["speed"] == "10000" and
+                    fvs["index"] == str(recirc_port_index_base + 1))
+
+        # Verify lanes in asicDB
+        asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+        asic_db_lanes_tbl = swsscommon.Table(asic_db, "LANES")
+
+        lanes = asic_db_lanes_tbl.get('')[1]
```

Comment:
use wait_for function to improve robustness

Reply:
Thanks for the suggestion. Fixed.